### PR TITLE
build(rust): add `rust-toolchain.toml`

### PR DESCRIPTION
**Pull request by rvolosatovs**
# Feature or Problem

`rust-toolchain.toml` is the standard Rust toolchain specification file, containing, among other things, the desired Rust toolchain channel, targets and components required for development

# Related Issues

# Release Information

# Consumer Impact

# Testing

Tested on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

## Platforms

Built on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[toolchain]
+channel = "stable"
+components = ["clippy", "rustfmt"]
+targets = [
+    "wasm32-unknown-unknown"
+]
```
Comment (brooksmtownsend) [inline rust-toolchain.toml:6]: I don't think this should be `wasm32-unknown-unknown`, unless I misunderstand the purpose of the targets. We build `wash` for a variety of Native targets but not for Wasm

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]

File: rust-toolchain.toml (hunk 1 above)
Comment (rvolosatovs) [reply rust-toolchain.toml:6]: I took it from https://github.com/wasmCloud/wash/pull/403/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R160

This does not specify the targets that `wash` is *built for*, but rather the targets that should be *available*

**Review by brooksmtownsend — APPROVED** [review]

File: rust-toolchain.toml (hunk 1 above)
Comment (brooksmtownsend) [reply rust-toolchain.toml:6]: Perfect, that's what I was concerned about

**Review by ghost — APPROVED** [review]
LGTM!

Also kinda unrelated but imagine if `nextest` could be added as a `component` here 🤔, would fix the concerns in #396

**Comment by rvolosatovs** [comment]
Looks like the CI still needs a push (same flake as in #407)

**Comment by thomastaylor312** [comment]
Because this is a known flake we are working on and this file doesn't touch code, I am going to merge as is

**Pull request merged**